tracer.go

```diff
@@ -14,13 +16,36 @@ type Tracer interface {
 	//
 	// Examples:
 	//
-	//     var tracer Tracer = ...
+	//     var tracer opentracing.Tracer = ...
 	//
-	//     sp := tracer.StartTrace("GetFeed")
+	//     sp := tracer.StartSpan("GetFeed")
 	//
-	//     sp := tracer.StartTrace("HandleHTTPRequest").
-	//         SetTag("user_agent", req.UserAgent).
-	//         SetTag("lucky_number", 42)
+	//     sp := tracer.StartSpanWithOptions(opentracing.SpanOptions{
+	//         OperationName: "LoggedHTTPRequest",
+	//         Tags: opentracing.Tags{"user_agent", loggedReq.UserAgent},
+	//         StartTime: loggedReq.Timestamp,
+	//     })
 	//
-	StartTrace(operationName string) Span
+	StartSpan(operationName string) Span
+	StartSpanWithOptions(opts *SpanOptions) Span
+}
+
+// StartSpanOptions allows Tracer.StartSpanWithOptions callers to override the
+// start timestamp, specify a parent Span, and make sure that Tags are
+// available at Span initialization time.
+type StartSpanOptions struct {
+	// OperationName may be empty (and set later via Span.SetOperationName)
+	OperationName string
+
+	// Parent may specify Span instance that caused the new (child) Span to be
+	// created. May be nil.
+	Parent Span
+
+	// StartTime overrides the Span's start time, or implicitly becomes
+	// time.Now() if StartTime.IsZero().
+	StartTime time.Time
```

Comment:
Have you considered not having a start- and end time for the trace and implicitly defining these two as the lowest and highest timestamp of the timestamps of the log events in a trace? Maybe there are compelling reasons against it, but it would remove these issues completely, without any interface changes.